AcsfToolsUtils.php

```diff
@@ -308,4 +264,117 @@ public function isOperationBlocked(array $data): bool {
+      if (!file_exists($sites_json_filepath)) {
+        $self = $this->siteAliasManager()->getSelf();
+        $process = Drush::drush($self, 'rsync', [$this->aliasRecord->name() . ':/mnt/files/' . $alias_name . '/files-private/sites.json',  $sites_json_filepath, '-y']);
+
+        try {
+          $process->mustRun();
+        }
+        catch (\Exception $e) {
+          return FALSE;
+        }
+      }
+
+      $json = @file_get_contents($sites_json_filepath);
+      $map = $json ? json_decode($json, TRUE) : FALSE;
+    }
+
+    // Look for list of sites and loop over it.
+    if ($map && isset($map['sites'])) {
+      // Acquire sites info.
+      $sites = array();
+      foreach ($map['sites'] as $domain => $site_details) {
+        if (!isset($sites[$site_details['name']])) {
+          $sites[$site_details['name']] = $site_details;
+        }
+
+        // Path domains need a trailing slash to be recognized as a drush alias.
+        if (FALSE !== strpos($domain, '/')) {
+          $domain = rtrim($domain, '/') . '/';
+        }
+
+        $sites[$site_details['name']]['domains'][] = $domain;
+
+        // Identify the site machine name from the acsitefactory.com domain.
+        $machine_name = [];
+        if (preg_match('/(.*)\..*\.acsitefactory\.com/', $domain, $machine_name)) {
+          $sites[$site_details['name']]['machine_name'] = $machine_name[1];
+        }
+      }
+    }
+    else {
+      $this->logger()->error("\nFailed to retrieve the list of sites of the factory.");
+    }
+
+    return $sites;
+  }
+
+  /**
+   * Utility function to retrieve a list of sites in a multi-site Drupal installation.
+   *
+   * @return array
+   * @throws \RuntimeException
+   */
+  public function getMultiSiteSites(): array {
+    // Read information from the sites.php file inside `/sites` directory.
+    // @todo: Change this to proper file where data is available.
+    $multisite_directory_path = DRUPAL_ROOT . '/sites';
+    $multisite_file_path = $multisite_directory_path.'/sites.php';
+
+    if (!is_file($multisite_file_path)) {
+      throw new \RuntimeException("Cannot find $multisite_file_path");
+    }
+
+    require $multisite_file_path;
+
+    if (!isset($sites) || !is_array($sites)) {
+      throw new \RuntimeException("Multi-site not defined in $multisite_file_path");
+    }
+
+    $site_details = [];
+    foreach ($sites as $site_name => $site_path) {
+      $site_details[$site_path] = [
+        'name' => $site_path,
+        'domains' => [
+          $site_name,
+        ],
+        'flags' => [],
+        'conf' => [
+          // These below array maintained same as ACSF sites.
+          'db_name' => $site_path, // Default to site name as DB name.
+          'site_id' => $site_path, // Default to site name as site ID.
+          'gardens_site_id' => $site_path, // Default to site name as gardens site ID.
+          'gardens_db_name' => $site_path, // Default to site name as gardens DB name.
```

Comment:
For now, I have kept the structure same as ACSF and also added this `garden_site_id` and `garden_db_name` as used  in helper method called in some classes https://github.com/acquia/acsf-tools/blob/11.x-dev/AcsfToolsCommands.php#L728